### ISSUE TITLE
Preserve source directory structure for chained targets

### ISSUE THREAD

**Issue by fatcerberus** [spheredev/neosphere#110]
Currently in master, the following Cellscript loses subdirectories:
```js
const transpile = require('transpile');
const minify = require('minify');
var sourceFiles = files('src/*.js', true);
var scripts = transpile('.tmp/', sourceFiles);
minify('@/scripts/', scripts);
```

The `minify()` step will produce a flat list of files in the output directory because the directory information is lost during the `transpile()` step.  Specifically, Target objects returned by `transpile()` don't contain any path information in their `.name` property, even if its sources do.  This should be fixed.

Fixing the bug will require a bit of refactoring to the API for targets.  `Target#name` may need to be made writable, for example.

**Comment by fatcerberus** [comment]
This shouldn't be too hard to fix, I'll just have to come up with an API for it.  The main problem is that `Tool#stage()` uses only the filename as the `.name` of the returned Target, because there's nothing else sensible it can do (targets don't track which directory they were ultimately sourced from).

It might be worth just adding an `options` parameter to `Tool#stage`.  This way you'd pass it the target filename, an array of Targets for the sources, plus an optional options object containing, e.g. the Target `.name`.  I'll look into it tomorrow.

**Comment by fatcerberus** [comment]
Fixed in 82fd29ac14f3ef038502ae2af5eae761ee5c7a80; subdirectory information is now preserved when passing files to `transpile()` or `minify()`.
